workflow/controller/operator.go

```diff
@@ -2624,3 +2646,60 @@ func (woc *wfOperationCtx) includeScriptOutput(nodeName, boundaryID string) (boo
 	}
 	return false, nil
 }
+
+func (woc *wfOperationCtx) getTopLevelWorkflowTemplate() (*wfv1.WorkflowSpec, error) {
+	var wftmpl wfv1.WorkflowSpecHolder
+	var err error
+	// Logic for Top level Workflow template reference
+	if woc.wf.Spec.WorkflowTemplateRef != nil {
+		if woc.wf.Spec.WorkflowTemplateRef.ClusterScope {
+			wftmpl, err = woc.controller.cwftmplInformer.Lister().Get(woc.wf.Spec.WorkflowTemplateRef.Name)
+		} else {
+			wftmpl, err = woc.controller.wftmplInformer.Lister().WorkflowTemplates(woc.wf.Namespace).Get(woc.wf.Spec.WorkflowTemplateRef.Name)
+
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return wftmpl.GetSpec().DeepCopy(), nil
+}
+
+func (woc *wfOperationCtx) convertWFTmplRefToTmplRef() *wfv1.TemplateRef {
+
+	entrypoint := woc.wf.Spec.Entrypoint
+
+	if entrypoint == "" {
+		entrypoint = woc.wfSpec.Entrypoint
+	}
+	return woc.wf.Spec.WorkflowTemplateRef.ConvertTemplateRef(entrypoint)
+}
+
+func (woc *wfOperationCtx) setWorkflowSpecAndEntrypoint() error {
+
+	woc.arguments.Parameters = woc.wf.Spec.Arguments.Parameters
+	woc.entrypoint = woc.wf.Spec.Entrypoint
+
+	if woc.wf.Spec.WorkflowTemplateRef == nil {
+		woc.wfSpec = woc.wf.Spec.DeepCopy()
+		return nil
+	}
+	if woc.wf.Status.StoredWorkflowTemplateSpec != nil {
```

Comment:
This should be an `else if`

Reply:
there is no statement after if statement for the first condition that's why there is a return statement inside block. I think `else if` is not needed